internal/priorityqueue/priorityqueue.go

```diff
@@ -44,10 +44,42 @@ func (pq *PriorityQueue[T]) PopMax() T {
 	maxItem := pq.items[1]
 	pq.items[1] = pq.items[len(pq.items)-1]
 	pq.items = pq.items[:len(pq.items)-1]
-	pq.siftdown()
+	pq.siftdown(1)
 	return maxItem
 }
 
+// RemoveFunc removes all elements for which rm returns true.
+func (pq *PriorityQueue[T]) RemoveFunc(rm func(T) bool) {
+	// This is effectively slices.DeleteFunc, but inlined because we start from index 1,
+	// and we don't need to clear the tail.
+	i := 1
+	for ; i < len(pq.items); i++ {
+		if rm(pq.items[i]) {
+			break
+		}
+	}
+	if i == len(pq.items) {
+		return // nothing to remove
+	}
+	for j := i + 1; j < len(pq.items); j++ {
+		v := pq.items[j]
```

Comment:
Nit: please rewrite this w/o continue, e.g. body in `if v := pq.items[j]; !rm(v) {`...

Reply:
as you wish

Reply:
Done